hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -282,16 +282,21 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(final Lon
         final List<String> targetIds = targetsWithActionType.stream().map(TargetWithActionType::getControllerId)
                 .distinct().collect(Collectors.toList());
 
+        final long notExistingTargets = targetIds.stream()
+                .filter(controllerId -> !targetRepository.existsByControllerId(controllerId)).count();
```

Comment:
The problem with this approach is that it will execute N database queries for each provided controllerId that could lead to increased latency. I would suggest filtering out non-existing controllerIds by repository itself, so at the end it looks like this:
```suggestion
                final List<String> existingTargetIds = Lists.partition(targetIds, Constants.MAX_ENTRIES_IN_STATEMENT)
                .stream().map(targetRepository::filterNonExistingControllerIds).flatMap(List::stream)
                .collect(Collectors.toList());
```
using it further and getting the total size from `existingTargetIds.size()`

The target repository implementation would then look like:
```suggestion
    @Query("SELECT t.controllerId FROM JpaTarget t WHERE t.controllerId IN ?1")
    List<String> filterNonExistingControllerIds(Iterable<String> controllerIds);
```

And finally the test should look similar to:
```suggestion
        final List<String> providedTargetIds = Lists.newArrayList("1", "notExistingTarget", "2");
        testdataFactory.createTarget("1");
        testdataFactory.createTarget("2");

        List<String> filteredIds = targetRepository.filterNonExistingControllerIds(providedTargetIds);
        
        assertThat(filteredIds).hasSize(2);
        assertThat(filteredIds).containsExactlyInAnyOrder("1", "2");
```